File: OneToOneCallSample/app/src/main/java/com/stringee/apptoappcallsample/listener/OnCallListener.java
package com.stringee.apptoappcallsample.listener;

import com.stringee.apptoappcallsample.common.CallStatus;
import com.stringee.video.StringeeVideoTrack;

public interface OnCallListener {
    void onCallStatus(CallStatus status);

    void onError(String message);

    void onReceiveLocalStream();

    void onReceiveRemoteStream();

    void onSpeakerChange(boolean isOn);

    void onMicChange(boolean isOn);

    void onVideoChange(boolean isOn);
    void onSharing(boolean isSharing);

    void onTimer(String duration);

    void onVideoTrackAdded(StringeeVideoTrack stringeeVideoTrack);

    void onVideoTrackRemoved(StringeeVideoTrack stringeeVideoTrack);
}

File: Kotlin-OneToOneCallSample/app/src/main/java/com/stringee/kotlin_onetoonecallsample/listener/OnCallListener.kt
package com.stringee.kotlin_onetoonecallsample.listener

import com.stringee.kotlin_onetoonecallsample.common.CallStatus
import com.stringee.video.StringeeVideoTrack


interface OnCallListener {
    fun onCallStatus(status: CallStatus)
    fun onError(message: String?)
    fun onReceiveLocalStream()
    fun onReceiveRemoteStream()
    fun onSpeakerChange(isOn: Boolean)
    fun onMicChange(isOn: Boolean)
    fun onVideoChange(isOn: Boolean)
    fun onSharing(isSharing: Boolean)
    fun onTimer(duration: String)
    fun onVideoTrackAdded(stringeeVideoTrack: StringeeVideoTrack)
    fun onVideoTrackRemoved(stringeeVideoTrack: StringeeVideoTrack)
}
